File: ruby/raindrops/raindrops.rb
# :nodoc
class Raindrops
  def self.m
    { 3 => 'Pling', 5 => 'Plang', 7 => 'Plong' }
  end

  def self.convert(n)
    s = (m.keys.map { |e| (n % e).zero? ? m[e] : '' }).join('')
    s != '' ? s : n.to_s
  end
end

File: go/raindrops/raindrops.go
// Package raindrops does a thing
package raindrops

import "fmt"

// Convert does a thing
func Convert(input int) string {
	res := ""
	if input%3 == 0 {
		res += "Pling"
	}
	if input%5 == 0 {
		res += "Plang"
	}
	if input%7 == 0 {
		res += "Plong"
	}
	if res != "" {
		return res
	}
	return fmt.Sprintf("%d", input)
}
